source/composer/midi_stream.cpp

```diff
@@ -0,0 +1,54 @@
+//
+// Created by Carla on 08/05/2021.
+//
+
+#include "midi_stream.h"
+#include <iostream>
+#include <sstream>
+
+auto midi_stream::convert_message(smf::MidiMessage &event) {
+
+    int x;
+    cycfi::q::midi::raw_message message{};
+    std::stringstream stream;
+
+    // Converts messages to hexadecimal
+    for(size_t i = event.size(); i > 0; --i){
+        stream << std::hex << static_cast<int>(event[i-1]);
+    }
+
+    stream >> x;
+
+    message.data = x;
+
+    return message;
+}
+
+bool midi_stream::is_harmony() {
+
+    for (int event = 1; event < event_list.size(); ++event) {
+        if (event_list[event][0] == note_on && event_list[event-1][0] == note_on) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void midi_stream::process(midi_processor &processor) {
+
+    // Compositions with harmonies will not be supported at first
+    if(is_harmony()){
+        std::cout << "This MIDI file contains harmonies." << std::endl;
```

Comment:
throw std::logic_error("This MIDI file contains harmonies: not implemented yet") ou std::runtime_error("This MIDI file contains harmonies: not implemented yet")